atom/browser/ui/file_dialog_mac.mm

```diff
@@ -93,6 +93,7 @@ void SetupDialog(NSSavePanel* dialog,
 
 void SetupDialogForProperties(NSOpenPanel* dialog, int properties) {
   [dialog setCanChooseFiles:(properties & FILE_DIALOG_OPEN_FILE)];
+  [dialog setTreatsFilePackagesAsDirectories:YES];
```

Comment:
@shubham2892 what do you think about making this an option instead of always on? Similar to how `setResolveAliases` support was added in #8617.